AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -723,11 +724,27 @@ private static void handleApiResponse(JSONObject response, List<Map<String, Obje
       for (int i = requests.size() - 1; i >= 0; i--) {
         Map<String, Object> currentRequest = requests.get(i);
         if (Constants.Methods.START.equals(currentRequest.get(Constants.Params.ACTION))) {
-          if (i < responseCount) {
-            lastStartResponse = Request.getResponseAt(response, i);
+          if (currentRequest.containsKey(Constants.Params.REQ_ID)) {
+              for (int j = Request.numResponses(response) - 1; j >= 0; j--) {
+                JSONObject currentResponse = Request.getResponseAt(response, i);
+                try {
+                  if (currentResponse.get(Constants.Params.REQ_ID) ==
+                          currentRequest.get(Constants.Params.REQ_ID)) {
+                    lastStartResponse = currentResponse;
+                    hasStartResponse = true;
+                    break;
+                  }
+                } catch (JSONException e) {
+                  Util.handleException(e);
+              }
+            }
+          } else {
```

Comment:
I think we should be handling backwards compatibility on the server, in case we introduce any bugs it'll be easier to fix. Thoughts?

Reply:
server will just pass through whatever `reqId` it gets, so that shouldnt be a problem either way... since the client will also use the older logic if the request id's dont match

Reply:
👍 so this code will never get executed right?

Reply:
yup, exactly.